src/Error/BaseErrorHandler.php

```diff
@@ -302,107 +312,36 @@ protected function _logError($level, array $data): bool
     }
 
     /**
-     * Handles exception logging
+     * Log an error for the exception if applicable.
      *
-     * @param \Throwable $exception Exception instance.
+     * @param \Throwable $exception The exception to log a message for.
+     * @param \Psr\Http\Message\ServerRequestInterface $request The current request.
      * @return bool
      */
-    protected function _logException(Throwable $exception): bool
+    public function logException(Throwable $exception, ?ServerRequestInterface $request = null): bool
     {
         $config = $this->_options;
         if (empty($config['log'])) {
             return false;
         }
 
-        if (!empty($config['skipLog'])) {
-            foreach ((array)$config['skipLog'] as $class) {
-                if ($exception instanceof $class) {
-                    return false;
-                }
-            }
-        }
-
-        return Log::error($this->_getMessage($exception));
-    }
-
-    /**
-     * Get the request context for an error/exception trace.
-     *
-     * @param \Cake\Http\ServerRequest $request The request to read from.
-     * @return string
-     */
-    protected function _requestContext(ServerRequest $request): string
-    {
-        $message = "\nRequest URL: " . $request->getRequestTarget();
-
-        $referer = $request->getEnv('HTTP_REFERER');
-        if ($referer) {
-            $message .= "\nReferer URL: " . $referer;
-        }
-        $clientIp = $request->clientIp();
-        if ($clientIp && $clientIp !== '::1') {
-            $message .= "\nClient IP: " . $clientIp;
-        }
-
-        return $message;
-    }
-
-    /**
-     * Generates a formatted error message
-     *
-     * @param \Throwable $exception Exception instance
-     * @return string Formatted message
-     */
-    protected function _getMessage(Throwable $exception): string
```

Comment:
We should note in the migration guide that the internals of Error handling have changed and that any sub-classes may need to be re-worked.